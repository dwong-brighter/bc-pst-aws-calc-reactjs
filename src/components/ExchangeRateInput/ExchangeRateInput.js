import React, { useState } from 'react';
import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import TextField from '@material-ui/core/TextField';

const useStyles = makeStyles({
  exchangeRateField: {
    margin: '0.5rem',
    maxWidth: '50%',
  }
});

/**
 * Formatter for a text field to become an exchange rate field.
 * 
 * @param {*} props 
 */
function NumberFormatExchangeRate (props) {
  const { foreignCurrencyCode, inputRef, onChange, ...other } = props;

  return (
    <NumberFormat
      {...other}
      getInputRef={inputRef}
      onValueChange={values => {
        onChange({
          target: {
            value: values.value,
          },
        });
      }}
      thousandSeparator
      decimalScale={13}
      fixedDecimalScale
      suffix={' CAD to 1 ' + foreignCurrencyCode}
    />
  );
}

NumberFormatExchangeRate.propTypes = {
  foreignCurrencyCode: PropTypes.string.isRequired,
  inputRef: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

function ExchangeRateInput(props) {
  const { 
    name,
    description,
    foreignCurrencyCode,
    initialRate,
    onRateChange
  } = props;

  const classes = useStyles();

  const [rate, setRate] = useState(initialRate);

  return (
    <TextField
      className={classes.exchangeRateField}
      label={description}
      value={rate}
      onChange={(event) => {
        let newRate = parseFloat(event.target.value);
        setRate(newRate); // update our state
        onRateChange(newRate); // update the app's state
      }}
      id={name}
      InputProps={{
        inputComponent: (props) =>
          <NumberFormatExchangeRate foreignCurrencyCode={foreignCurrencyCode} {...props} />,
      }}
    />
  );
}

ExchangeRateInput.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  foreignCurrencyCode: PropTypes.string.isRequired,
  initialRate: PropTypes.number.isRequired,
  onRateChange: PropTypes.func.isRequired,
};

export default ExchangeRateInput;