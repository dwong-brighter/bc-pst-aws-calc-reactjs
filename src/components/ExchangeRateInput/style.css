.ExchangeRateInput-container {
  display: flex;
}

.ExchangeRateInput-value {
  border-radius: 5px;
  border-width: 0;
  font-family: sans-serif;
  margin: 5px 10px;
  padding: 5px;
}
