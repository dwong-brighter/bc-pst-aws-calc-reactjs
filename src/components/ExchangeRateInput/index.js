import ExchangeRateInput from './ExchangeRateInput.js';

export default ExchangeRateInput;
