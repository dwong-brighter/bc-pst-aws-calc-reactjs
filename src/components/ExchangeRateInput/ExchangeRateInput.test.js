import React from 'react';
import { shallow } from 'enzyme';
import ExchangeRateInput from './ExchangeRateInput';

describe('<ExchangeRateInput />', () => {
  test('renders', () => {
    const wrapper = shallow(
      <ExchangeRateInput
        name="someRate"
        description="some exchange rate"
        foreignCurrencyCode="USD"
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
