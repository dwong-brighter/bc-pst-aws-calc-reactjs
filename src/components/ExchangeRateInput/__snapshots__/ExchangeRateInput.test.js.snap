// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeRateInput /> renders without crashing 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "inputComponent": [Function],
    }
  }
  className="makeStyles-exchangeRateField-1"
  id="someRate"
  label="some exchange rate"
  onChange={[Function]}
  value={4.2}
/>
`;
